Engine/Component/ComponentAnimation.cpp

```diff
@@ -0,0 +1,131 @@
+#include "ComponentAnimation.h"
+
+#include "Animation/AnimController.h"
+#include "Component/ComponentTransform.h"
+#include "Main/Application.h"
+#include "Main/GameObject.h"
+#include "Module/ModuleAnimation.h"
+#include "Module/ModuleTime.h"
+#include "Module/ModuleResourceManager.h"
+
+ComponentAnimation::ComponentAnimation() : Component(nullptr, ComponentType::ANIMATION)
+{
+	animation_controller = new AnimController();
+	SetAnimation(App->resources->Load<Animation>("Library/Metadata/38/3800295065"));
+	SetSkeleton(App->resources->Load<Skeleton>("Library/Metadata/29/2987806620"));
+}
+
+ComponentAnimation::ComponentAnimation(GameObject * owner) : Component(owner, ComponentType::ANIMATION)
+{
+	animation_controller = new AnimController();
+}
+
+ComponentAnimation::~ComponentAnimation()
+{
+}
+
+Component* ComponentAnimation::Clone(bool original_prefab) const
+{
+	ComponentAnimation * created_component;
+	if (original_prefab)
+	{
+		created_component = new ComponentAnimation();
+	}
+	else
+	{
+		created_component = App->animations->CreateComponentAnimation();
+	}
+	*created_component = *this;
+	return created_component;
+};
+
+void ComponentAnimation::Copy(Component * component_to_copy) const
+{
+	*component_to_copy = *this;
+	*static_cast<ComponentAnimation*>(component_to_copy) = *this;
+}
+
+void ComponentAnimation::Render(GLuint shader_program) 
+{
+	glUniformMatrix4fv(glGetUniformLocation(shader_program, "palette"), palette.size(), GL_TRUE, &palette[0][0][0]);
+}
+void ComponentAnimation::Update()
+{
+	animation_controller->Update();
+
+	if (animation_controller->playing)
+	{
+		UpdateBone(owner);
+	}
+}
+
+void ComponentAnimation::Delete()
+{
+	App->animations->RemoveComponentAnimation(this);
+}
+
+void ComponentAnimation::Save(Config& config) const
+{
+	config.AddUInt(UUID, "UUID");
+	config.AddUInt((uint64_t)type, "ComponentType");
+	config.AddBool(active, "Active");
+
+
+	config.AddString(animation_controller->anim->exported_file, "AnimationResource");
+	config.AddString(animation_controller->sk->exported_file, "SkeletonResource");
+}
+
+void ComponentAnimation::Load(const Config& config)
+{
+	UUID = config.GetUInt("UUID", 0);
+	active = config.GetBool("Active", true);
+	std::string animation_path;
+	config.GetString("AnimationResource", animation_path, "");
+	SetAnimation(App->resources->Load<Animation>(animation_path));
+
+	std::string skeleton_path;
+	config.GetString("SkeletonResource", skeleton_path, "");
+	SetSkeleton(App->resources->Load<Skeleton>(skeleton_path));
+}
+
+void ComponentAnimation::SetAnimation(std::shared_ptr<Animation> & animation)
```

Comment:
code style, sorry you are going to hate me but.. blame on @mariofv and @Dartemiss